lib/hookRunner.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+function hookRunner (functions, context) {
+  functions = functions.map(fn => fn.bind(context))
+
+  return function runHooks (runner, state, cb) {
+    var i = 0
+
+    function next (err, value) {
+      if (err) {
+        cb(err, state)
```

Comment:
you would need use `call` here, as we maintain the context as this.

Reply:
That doesn't seem to be necessary since none of the hook callbacks use the `this` keyword:

+ [onRequest](https://github.com/fastify/fastify/blob/f575eba8a1d3b0a30db35a7e14a25a2d94791023/fastify.js#L292-L307)
+ [preHandler](https://github.com/fastify/fastify/blob/f575eba8a1d3b0a30db35a7e14a25a2d94791023/lib/handleRequest.js#L156-L177)
+ [onSend](https://github.com/fastify/fastify/blob/f575eba8a1d3b0a30db35a7e14a25a2d94791023/lib/reply.js#L136-L153) (`this` is used here, but it's bound to `reply`)
+ [onResponse](https://github.com/fastify/fastify/blob/f575eba8a1d3b0a30db35a7e14a25a2d94791023/lib/logger.js#L67-L83)

Reply:
ok amazing!